### PR TITLE
Ring signature support on Azimuth points

### PR DESCRIPTION
**Pull request by eglaysher**
This adds `%/lib/ring/hoon` which provides both a low-level raw ring signature implementation and a higher level one which understands urbit nouns and azimuth points.

Using this library, the following minimal test generator will build a linked ring signature with your private key while hiding in the set of identities passed in:

```
/+  ring
::
:-  %say
|=  [{now/@da eny/@uvJ bec/beak} [invited=(list @p) ~] ~]
:-  %noun
=/  s1  (sign:ring p.bec now eny "Hello World" `[%scope 6] (sy invited))
~&  [%signature-1 s1]
=/  v  (verify:ring p.bec now "Hello World" s1)
v
```

This patch requires the jets in https://github.com/urbit/urbit/pull/1300 to be viable for any real use: without the jets, creating a linked ring signature with three participants is a minute and a half, but is instantaneous with the jets.

### CODE REVIEW

**Comment by philipcmonk** [comment]
Can you confirm that `+test` works?  Travis is killing the build because it's stalling, presumably because it's not using the jets.

**Comment by eglaysher** [comment]
Just ran the entire suite locally and can confirm it passed. Should I wait for the jets to land then?

File: lib/ring.hoon [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+    (build-signing-participants:detail our now ~(tap in anonymity-set))
+  ::  get our ships' current life
+  ::
+  =/  our-life=life
+    .^(life %j /(scot %p our)/life/(scot %da now)/(scot %p our))
+  ::  get our ships' secret keyfile ring
+  ::
+  =/  secret-ring=ring
+    .^(ring %j /(scot %p our)/vein/(scot %da now)/(scot %ud our-life))
+  ::  fetch the encoded auth seedd from the ring
```
Comment (philipcmonk) [inline lib/ring.hoon]: dd

**Comment by philipcmonk** [comment]
We can't break CI on master, so yeah, I'd say make sure the jets are merged first.

**Comment by jtobin** [comment]
(Re: getting the jets into master on vere, see my comments on the matching PR there.)

**Comment by joemfb** [comment]
FYI, you can set the commit that will be used to build vere in CI in `.travis/pin-vere-commit.txt`. That needn't point to master, just any commit/tag/branch in the `urbit/urbit` repository.

**Comment by eglaysher** [comment]
PTAL. I have disabled the tests in this patch so it can be merged into master. Once we've merged to the monorepo with cc-release, we can reenable them.

**Comment by eglaysher** [comment]
Since I have two approvals, and the CI passes, I'm just going to merge this.

**Pull request merged**